### PR TITLE
Microservices liveness probes with help of Prometheus 

### PR DESCRIPTION
**Pull request by TheRealSpaceShip**
This PR contains manifest files for the setup of the microservices liveness probes with Prometheus (OSS and GMP)

* Configure the liveness probes with help of Blackbox exporter probes
* Configure Alert rules
* Configure Alertmanager to send notifications via Slack channel

### CODE REVIEW

**Comment by TheRealSpaceShip** [comment]
Yes, please, review it.

File: extras/prometheus/gmp/README.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# GMP on GKE cluster
+
+This directory contains files related to configuring probing of Bank of Anthos' microservices with GMP on GKE
+
+* `blackbox-exporter.yaml` - Deployment manifest of [Blackbox exporter](https://github.com/prometheus/blackbox_exporter/)
+* `probes.yaml` - File contains GMP PodMonitoring custom resource which scrape metrics from Blackbox Exporter with configuration for each microservice.
```
Comment (Shabirmean) [inline extras/prometheus/gmp/README.md:6]:
```suggestion
* `probes.yaml` - contains the GMP PodMonitoring custom resource definitions per microservice. One monitoring configuration per microservice is defined to be run by the  Blackbox exporter.
```

Comment (Shabirmean) [reply extras/prometheus/gmp/README.md:6]: Stay consistent with the naming of `Blackbox exporter`. Uppercase B and lowercase e

File: extras/prometheus/gmp/README.md [hunk 2]
```diff
@@ -0,0 +1,8 @@
+# GMP on GKE cluster
+
+This directory contains files related to configuring probing of Bank of Anthos' microservices with GMP on GKE
+
+* `blackbox-exporter.yaml` - Deployment manifest of [Blackbox exporter](https://github.com/prometheus/blackbox_exporter/)
```
Comment (Shabirmean) [inline extras/prometheus/gmp/README.md:5]:
```suggestion
* `blackbox-exporter.yaml` - contains the Deployment definition of [Blackbox exporter](https://github.com/prometheus/blackbox_exporter/)
```

File: extras/prometheus/gmp/README.md [hunk 3]
```diff
@@ -0,0 +1,8 @@
...
+* `probes.yaml` - File contains GMP PodMonitoring custom resource which scrape metrics from Blackbox Exporter with configuration for each microservice.
+* `rules.yaml` - File contains GMP Rules custom resource
```
Comment (Shabirmean) [inline extras/prometheus/gmp/README.md:7]:
```suggestion
* `rules.yaml` - contains the GMP Rules custom resource definition.
```

File: extras/prometheus/gmp/README.md [hunk 4]
```diff
@@ -0,0 +1,8 @@
...
+* `alertmanager.yaml` - File contains configuration of Alertmanager to send notifications to Slack channel via Webhook URL.
```
Comment (Shabirmean) [inline extras/prometheus/gmp/README.md:8]:
```suggestion
* `alertmanager.yaml` - contains configuration of Alertmanager to send notifications to Slack channel via Webhook URL.
```

File: extras/prometheus/gmp/alertmanager.yaml [hunk 5]
```diff
@@ -0,0 +1,60 @@
+# Copyright 2022 Google LLC
```
Comment (Shabirmean) [inline extras/prometheus/gmp/alertmanager.yaml:1]: Copyright year 2023?

File: extras/prometheus/gmp/rules.yaml [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+spec:
+  groups:
+  - name: Micro services uptime
+    interval: 60s
+    rules:
+    - alert: BalancereaderUnavaiable
+      expr: probe_success{job="balancereader-probe"} == 0
+      for: 1m
+      annotations:
+        summary: Balance Reader service is unavailable
```
Comment (Shabirmean) [inline extras/prometheus/gmp/rules.yaml:28]: Follow the same naming convention for the microservices. i.e. Balance-reader as one word! Same for Ledger-writer. Check others as well

File: extras/prometheus/gmp/rules.yaml [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+      annotations:
+        summary: Transaction History service is unavailable
+        description: Check Transaction History pods and it's logs
+      labels:
+        severity: 'critical'
+    - alert: UserserviceUnavaiable
+      expr: probe_success{job="userservice-probe"} == 0
+      for: 1m
+      annotations:
+        summary: User Service is unavailable
```
Comment (Shabirmean) [inline extras/prometheus/gmp/rules.yaml:68]: `Service` -> `service`. Have it consistently simple case

File: extras/prometheus/oss/README.md [hunk 8]
```diff
@@ -0,0 +1,8 @@
+# OSS Prometheus on GKE cluster
```
Comment (Shabirmean) [inline extras/prometheus/oss/README.md:1]:
```suggestion
# OSS Prometheus on a GKE cluster
```

File: extras/prometheus/oss/README.md [hunk 9]
```diff
@@ -0,0 +1,8 @@
...
+
+This directory contains files related to configuring probing of Bank of Anthos' microservices with OSS Prometheus on GKE.
```
Comment (Shabirmean) [inline extras/prometheus/oss/README.md:3]:
```suggestion
This directory contains files essential to setup liveness probes for Bank of Anthos' microservices running in GKE using OSS Prometheus.
```

File: extras/prometheus/oss/README.md [hunk 10]
```diff
@@ -0,0 +1,8 @@
...
+
+* `values.yaml` - Configuration file to [OSS Prometheus helm chart](https://github.com/bitnami/charts/tree/main/bitnami/kube-prometheus)
```
Comment (Shabirmean) [inline extras/prometheus/oss/README.md:5]:
```suggestion
* `values.yaml` - contains the configurations for the [OSS Prometheus helm chart](https://github.com/bitnami/charts/tree/main/bitnami/kube-prometheus)
```

File: extras/prometheus/oss/README.md [hunk 11]
```diff
@@ -0,0 +1,8 @@
...
+* `probes.yaml` - File contains Probe custom resource manifests for each microservice
```
Comment (Shabirmean) [inline extras/prometheus/oss/README.md:6]:
```suggestion
* `probes.yaml` - contains the Probe custom resource definitions per microservice. One monitoring configuration per microservice is defined.
```

File: extras/prometheus/oss/README.md [hunk 12]
```diff
@@ -0,0 +1,8 @@
...
+* `rules.yaml` - File contains PrometheusRule custom resource
```
Comment (Shabirmean) [inline extras/prometheus/oss/README.md:7]:
```suggestion
* `rules.yaml` - contains the PrometheusRule custom resource definition.
```

File: extras/prometheus/oss/README.md [hunk 13]
```diff
@@ -0,0 +1,8 @@
...
+* `alertmanagerconfig.yaml` - File contain AlertmanagerConfig custom resource with configuration of Alertmanager to send notifications to Slack channel via Webhook URL.
```
Comment (Shabirmean) [inline extras/prometheus/oss/README.md:8]:
```suggestion
* `alertmanager.yaml` - contains configuration of AlertmanagerConfig custom resource . This resource defines the configurations necessary to send notifications to a Slack channel via Webhook URL.
```

File: extras/prometheus/oss/alertmanagerconfig.yaml [hunk 14]
```diff
@@ -0,0 +1,67 @@
+# Copyright 2022 Google LLC
```
Comment (Shabirmean) [inline extras/prometheus/oss/alertmanagerconfig.yaml:1]: Fix copyright years

File: extras/prometheus/oss/rules.yaml [hunk 15]
```diff
@@ -0,0 +1,71 @@
...
+spec:
+  groups:
+  - name: Micro services uptime
+    interval: 60s
+    rules:
+    - alert: BalancereaderUnavaiable
+      expr: probe_success{app="bank-of-anthos",job="balancereader"} == 0
+      for: 1m
+      annotations:
+        summary: Balance Reader service is unavailable
```
Comment (Shabirmean) [inline extras/prometheus/oss/rules.yaml:28]: Fix same comments as the other folder

**Review by Shabirmean — CHANGES_REQUESTED** [review]
Thank you for these improvements @TheRealSpaceShip . Very much appreciate the effort here. Please TAL at the comments.  Also please update your branch with the latest changes from the main branch!

File: extras/prometheus/gmp/rules.yaml (hunk 6 above)
Comment (TheRealSpaceShip) [reply extras/prometheus/gmp/rules.yaml:28]: Naming was done the same way as declared in the `README.md` of each service.

File: extras/prometheus/gmp/rules.yaml (hunk 7 above)
Comment (TheRealSpaceShip) [reply extras/prometheus/gmp/rules.yaml:68]: Updated name the same way as it is declared in the `README.md` files for each microservice.

File: extras/prometheus/gmp/README.md [hunk 16]
```diff
@@ -0,0 +1,8 @@
+# GMP on GKE cluster
+
+This directory contains files related to configuring probing of Bank of Anthos' microservices with GMP on GKE
+
```
[the remark this answers is not in the document]
Comment (TheRealSpaceShip) [reply extras/prometheus/gmp/README.md:4]: Got it. Thank you.

**Review by Shabirmean — COMMENTED** [review]
Thanks for adopting the changes. 🚀 

PR wise it looks good to me. I'll let @bourgeoisor make the final review on approving and merging this!

**Comment by TheRealSpaceShip** [comment]
@bourgeoisor done. I have added short instructions to both READMEs.

**Review by bourgeoisor — APPROVED** [review]

**Pull request merged**